### PR TITLE
feat: [sc-154155] update Lua SDK hello app readme and comments to standard design

### PR DESCRIPTION
**Pull request by ember-stevens**
After receiving some feedback that it'd be helpful if the Hello apps were more standardized, Docs team is working on updating them to [match the spec](https://launchdarkly.atlassian.net/wiki/spaces/ENG/pages/1499562073/SDK+hello+world).

For this app, some readme updates and adding standard comments.

Story details: https://app.shortcut.com/launchdarkly/story/154155

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,16 @@
-# LaunchDarkly Lua Server-Side SDK HAProxy example
+# LaunchDarkly Lua server-side SDK HAProxy example
 
-This is a minimal dockerized example of using the Lua SDK with [HAProxy](https://www.haproxy.org/).
+We've built a minimal dockerized example of using the Lua SDK with [HAProxy](https://www.haproxy.org/). For more comprehensive instructions, you can visit the [Using the Lua SDK with HAProxy guide](https://docs.launchdarkly.com/guides/sdk/haproxy) or the [Lua reference guide](https://docs.launchdarkly.com/sdk/server-side/lua).
```
Comment (ember-stevens) [inline README.md:3]: @mmrj should we still link to tha API docs here?

**Comment by kinyoklion** [comment]
@cwaldren-ld Looks like this CI config needed updated for the CMake version update for c server?

File: README.md (hunk 1 above)
Comment (mmrj) [reply README.md:3]: yeah, linking to the SDK reference guide makes sense to me

**Review by mmrj — APPROVED** [review]

**Comment by cwaldren-ld** [comment]
Yes, will fix.

**Pull request merged**